### PR TITLE
chore: add balances for rockwallet pro

### PR DESCRIPTION
**Pull request by dangelovska**

### CODE REVIEW

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 1]
```diff
@@ -572,12 +572,35 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber,
...
+        var proBalance: Decimal = 0
+        switch code {
+        case Constant.BSV:
+            proBalance = proBalancesData?.bsv ?? 0
+        case Constant.ETH:
+            proBalance = proBalancesData?.eth ?? 0
+        case Constant.BTC:
+            proBalance = proBalancesData?.btc ?? 0
+        case Constant.USDC:
+            proBalance = proBalancesData?.usdc ?? 0
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:598]: empty line after cases

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 2]
```diff
@@ -572,12 +572,35 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber,
...
               let fiatCurrency = Store.state.orderedWallets.first?.currentRate?.code else { return }
-        totalAssetsAmountLabel.text = String(format: "\(Constant.currencyFormat)", formattedBalance, fiatCurrency)
+        totalAssetsAmountLabel.text = String(format: "%@ %@", formattedBalance, fiatCurrency)
+    }
+    
+    func getProBalance(code: String, proBalancesData: ProBalancesModel?) -> Decimal {
+        var proBalance: Decimal = 0
+        switch code {
+        case Constant.BSV:
+            proBalance = proBalancesData?.bsv ?? 0
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:592]: why default to 0 here again? you do it in the worker already

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 3]
```diff
@@ -572,12 +572,35 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber,
...
+            let amount = Amount(amount: proBalance,
+                                rate: $0.currentRate)
+            return amount.fiatValue
+        }.reduce(0.0, +)
         
-        guard let formattedBalance = ExchangeFormatter.fiat.string(for: balance),
+        guard let formattedBalance = ExchangeFormatter.fiat.string(for: fiatTotal),
               let fiatCurrency = Store.state.orderedWallets.first?.currentRate?.code else { return }
-        totalAssetsAmountLabel.text = String(format: "\(Constant.currencyFormat)", formattedBalance, fiatCurrency)
+        totalAssetsAmountLabel.text = String(format: "%@ %@", formattedBalance, fiatCurrency)
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:585]: why "%@ %@"? what was wrong with Constant.currencyFormat

File: rockwallet/rockwallet/src/ViewModels/HomeScreenAssetViewModel.swift [hunk 4]
```diff
@@ -30,4 +31,39 @@ struct HomeScreenAssetViewModel {
...
+            proBalance = proBalancesData?.bsv ?? 0
+        case Constant.ETH:
+            proBalance = proBalancesData?.eth ?? 0
+        case Constant.BTC:
+            proBalance = proBalancesData?.btc ?? 0
+        case Constant.USDC:
+            proBalance = proBalancesData?.usdc ?? 0
+        default:
+            break
+        }
```
Comment (kenanmamedoff) [inline rockwallet/rockwallet/src/ViewModels/HomeScreenAssetViewModel.swift:65]: same

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift (hunk 2 above)
Comment (dgacevic) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:592]: No need to set value to a temp property, just return in each case.

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 5]
```diff
@@ -416,7 +416,8 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber,
         if selectedSegment == .rockWalletPro {
             ProBalancesWorker().execute(requestData: ProBalancesRequestData()) { result in
                 switch result {
-                case .success(let data):
+                case .success(let responseData):
+                    guard let data = responseData else { return }
```
Comment (dgacevic) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:420]: `guard let responseData else { return }`

File: rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift [hunk 6]
```diff
@@ -582,25 +583,26 @@ class HomeScreenViewController: UIViewController, UITabBarDelegate, Subscriber,
...
-            proBalance = proBalancesData?.btc ?? 0
+            return proBalancesData.btc
+            
         case Constant.USDC:
-            proBalance = proBalancesData?.usdc ?? 0
+            return proBalancesData.usdc
+            
         default:
-            break
+            return 0
```
Comment (dgacevic) [inline rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:604]: Would probably be better to return nil instead, no?

Comment (kenanmamedoff) [reply rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:604]: Nil decimal... you will have to unwrap it somewhere probably.

Comment (dangelovska) [reply rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:604]: it's 0 because return type is Decimal.

Comment (dgacevic) [reply rockwallet/rockwallet/src/ViewControllers/HomeScreenViewController.swift:604]: Change `Decimal` to `Decimal?`

**Review by dgacevic — APPROVED** [review]

**Pull request merged**